File: src/service/emp.service.js
import axios from "axios";
const BASE_API_URL = "http://localhost:8080/api/v1";

class EmpService{
    saveEmp(emp)
    {
        return axios.post(BASE_API_URL + "/save", emp);
    }
    getAllEmp() {
        return axios.get(BASE_API_URL + "/");
    }

    getEmpById(id) {
        return axios.get(BASE_API_URL + "/" + id);
    }

    deleteEmp(id) {
        return axios.get(BASE_API_URL + "/delete/" + id);
    }

    updateEmp(id, emp) {
        return axios.post(BASE_API_URL + "/update/" + id, emp);
    }
}
export default new EmpService();

File: src/main/java/com/jalindar/service/EmpService.java
package com.jalindar.service;

import java.util.List;

import com.jalindar.model.Emp;

public interface EmpService {
	public Emp createEmp(Emp emp);

	public List<Emp> getAllEmp();

	public Emp getEmpById(int id);

	public void deleteEmp(int id);

	public Emp updateEmp(int id, Emp emp);

}
